processing/src/test/java/org/apache/druid/query/aggregation/AggregationTestHelper.java

```diff
@@ -509,50 +507,6 @@ public void createIndex(
     }
   }
 
-  public void createIndex(
-      InputStream inputDataStream,
-      String parserJson,
-      String transformSpecJson,
-      String aggregators,
-      File outDir,
-      long minTimestamp,
-      Granularity gran,
-      int maxRowCount,
-      boolean rollup
-  ) throws Exception
-  {
-    try {
-      StringInputRowParser parser = mapper.readValue(parserJson, StringInputRowParser.class);
-      TransformSpec transformSpec;
-      if (transformSpecJson != null) {
-        transformSpec = mapper.readValue(transformSpecJson, TransformSpec.class);
-        parser = new TransformingStringInputRowParser(parser.getParseSpec(), parser.getEncoding(), transformSpec);
-      }
-
-      LineIterator iter = IOUtils.lineIterator(inputDataStream, "UTF-8");
-      List<AggregatorFactory> aggregatorSpecs = mapper.readValue(
-          aggregators,
-          new TypeReference<List<AggregatorFactory>>()
-          {
-          }
-      );
-
-      createIndex(
-          iter,
-          parser,
-          aggregatorSpecs.toArray(new AggregatorFactory[0]),
-          outDir,
-          minTimestamp,
-          gran,
-          true,
-          maxRowCount,
-          rollup
-      );
-    }
-    finally {
-      Closeables.close(inputDataStream, true);
-    }
-  }
```

Comment:
Removing this method could quite easily break extensions.  IIRC, the `AggregationTestHelper` exists primarily to aid in building tests for AggregatorFactory extensions.  Did the same logic move somewhere else? If so, can we keep this method (maybe deprecated?) and have it call that other place?

Reply:
Hmm, I added this one in the PR that introduced nested columns. The same functionality that many of the methods that create segments in `AggregationTestHelper ` have is done better in `IndexBuilder`, which is why I focused on improving it and switching nested column tests to using it.

There were no users of this method after i refactored my tests so I removed it. I can always add it back if you're worried about it, but imo everyone should use `IndexBuilder` and we should deprecate most of `AggregationTestHelper`, at least the parts around making segments. I didn't not migrate the other users away from it in this PR though.